test/ILLink.RoslynAnalyzer.Tests/AvoidAssemblyLocationInSingleFileTests.cs

```diff
@@ -19,6 +19,33 @@ static Task VerifySingleFileAnalyzer (string source, params DiagnosticResult[] e
 				expected);
 		}
 
+		// TODO: Remove this once we have the new attribute available for linker testing.
+		static Task VerifySingleFileAnalyzerWithRequiresAssemblyFilesAttribute (string source, params DiagnosticResult[] expected)
+		{
+			source = @"
+using System.Reflection;
+namespace System.Diagnostics.CodeAnalysis
+{
+#nullable enable
+    [AttributeUsage(AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Property,
+                    Inherited = false,
+                    AllowMultiple = false)]
+    public sealed class RequiresAssemblyFilesAttribute : Attribute
```

Comment:
@vitek-karas I thought that we had the attribute already available since we already bump the linker to net6.0, but if I delete this block of code Roslyn will complain about System.Diagnostics.CodeAnalysis not having RequiresAssemblyFilesAttribute. I wasn't sure if I was missing anything 😵

Reply:
It is already checked-in in dotnet/runtime and I can see it in System.Private.CoreLib.dll; not sure how the analyzer resolves it though, I think @agocke might have an idea.

Reply:
I opened #1922, which should solve this, PTAL. When compiling the analyzer tests, we use .NET 5 assemblies, see:
https://github.com/mono/linker/blob/000432728ae8603d7ac0519d844bc260eee4910e/test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier%601.cs#L53

This is the reason why `RequiresAssemblyFilesAttribute` can't be used.